docs/integration/KafkaIntegration.md

```diff
@@ -8,104 +8,164 @@ sidebar_position: 1
+In Nussknacker terminology input topics are handled by *source* components, output topics are handled by *sink* components.
+This section provides important details of Nussknacker's integration with Kafka and Schema Registry.
 
-## Schema Registry integration
+### Schemas
 
-Nussknacker integrates with Schema Registries. It is the source of knowledge about:
-- Which topics are available in Kafka sources and sinks
-- What schema versions are available for topic
-- How code completions and validations on types described by schemas should work
-- How messages will be serialized and deserialized
+Nussknacker uses schemas for code completions and validations of messages. 
+Schema of message can be described in [Avro Schema format](https://avro.apache.org/docs/#schemas) or [JSON Schema format](https://json-schema.org) (Confluent Schema Registry only)
 
-Currently, Nussknacker supports two implementations of schema registries: based on *Confluent Schema Registry* and based on *Azure Schema Registry*. 
-During runtime, it serializes and deserializes messages in the way that is compatible with standard Kafka serializers and deserializers 
-delivered by those schema registries providers. Thanks to that you should be able to send messages to Nussknacker and read messages 
-produced by Nussknacker using standard tooling available around those Schema Registries.
+Schemas are retrieved from Schema Registry - *Confluent Schema Registry* and *Azure Schema Registry* are supported.
 
-Given implementation is picked based on `schema.registry.url` property. By default, Confluent-based implementation is used. 
-For urls ended with `.servicebus.windows.net` Azure-based implementation is used.
+To preview schemas or add a new version, you can use tools available on your cloud platform or tools like [AKHQ](https://akhq.io)
+
+#### Association between schema with topic
+
+To properly present information about topics and version and to recognize which schema is assigned to version, Nussknacker follow conventions:
+- For Confluent-based implementation it uses [TopicNameStrategy for subject names](https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#subject-name-strategy).
+  It means that it looks for schemas available at `<topic-name>-(key or value)` *subject*. For example for topic `transactions`, it looks for schemas at `transactions-key` *subject* for key and `transactions-value` *subject* for value
+- In the Azure Schema Registry, *subject* concept doesn't exist - schemas are grouped by the same *schema name*. Because of that, Nussknacker introduces
+  own convention for association between schema and topic: *schema name* should be in format: `CamelCasedTopicNameKey` for keys and `CamelCasedTopicNameValue` for values.
+  For example for `input-events` topic, schema name should be named `InputEventsKey` for key or `InputEventsValue` for value. Be aware that it may require change of schema name
```

Comment:
_Be aware that ..._
If the intention is to say that _if before you  implemented Nu you already had had schema definitions and they do not conform to our convention, Nu will not be happy_ - I think it is obvious and can be omitted. If the intention was different, I did not understand it.

Reply:
I think that it is impossible to understand this notice without usage of Azure portal. There are schema names that are something different than schema name inside schema. I can show you if you want.